DSCResources/MSFT_xWaitForADDomain/MSFT_xWaitForADDomain.psm1

```diff
@@ -42,18 +47,17 @@ function Set-TargetResource
     )
 
     $domainFound = $false
-    Write-Verbose -Message "Checking for domain $DomainName ..."
-
+    
     for($count = 0; $count -lt $RetryCount; $count++)
     {
-        try
+        $domain = Get-Domain @PSBoundParameters
```

Comment:
I think it would be a best practice to use Set-StrictMode -Version latest
and line 54 would fail with this on based on the behavior you described.  But, I would deal with this is a separate PR.